src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameGapCheck.java

```diff
@@ -184,6 +180,29 @@ private boolean findMatchingEdgeNameWithConnectedEdges(final Set<Edge> connected
                 && connectedEdge.getName().get().equals(edgeName));
     }
 
+    private FeatureChange getComplexFixSuggestion(final AtlasObject object,
+            final String originalName, final String nameSuggestion)
+    {
+        final CompleteEntity completeEntity = (CompleteEntity) CompleteEntity
+                .from((AtlasEntity) object);
+        completeEntity.withReplacedTag(NameTag.KEY, NameTag.KEY, nameSuggestion);
+        if (BridgeTag.isBridge(object) && object.getTag(BridgeNameTag.KEY).isEmpty())
+        {
+            completeEntity.withAddedTag(BridgeNameTag.KEY, originalName);
+        }
+        else
```

Comment:
I hate to say this, but for completeness should we also check for tunnels? In some parts of the world tunnels are named too...
```suggestion
        else if (TunnelTag.isTunnel(object) && object.getTag(TunnelNameTag.KEY).isEmpty())
        {
            completeEntity.withAddedTag(TunnelNameTag.KEY, originalName);
        }
        else
```